### PR TITLE
Refactor/accessibilty enhancements

### LINKED ISSUE #71: Accessibility: Change search input type to "search" instead of "text" and figure out label solution

### LINKED ISSUE #72: Accessibility: Adjust header level within recipe cards

### LINKED ISSUE #87: Truncate the recipe card title text.
Set each card title to a maximum length with a ... for the remainder. This will allow a standardized text box and text size.

### LINKED ISSUE #88: Create hover state to see full title without clicking into the recipe modal.
Requires truncated recipe titles first.

### PR DESCRIPTION
**Pull request by qrispi**
Files Changed: index.html, styles.css, scripts.js
Change Summary: Search bar input now has the proper input type as well as a hidden (by this I mean absolutely placed way off the page) label for screen readers. Recipe card titles are now uniform sized at 1.6rem, use semantic h2's, and get broken off by an ellipsis if they are too long. Padding, sizing and spacing for the recipe cards was adjusted to match. A hover class was also created and conditionally applied to titles over 18 chars to expand the title box with a scroll bar so the full title may be viewed. Text-box hover state styling matches the styling  of the recipe card to blend in. Alt-text for the recipe card images was also updated to not be 'suspicious'. There are no current issues or errors upon running the WAVE tool now.

Closes #88, Closes #87, Closes #72, Closes #71

### CODE REVIEW

**Comment by qrispi** [comment]
I am more than happy to demonstrate this functionality before the merge is completed!

File: dist/index.html [hunk 1]
```diff
@@ -11,7 +11,8 @@
     <nav class="nav-bar">
         <img class="nav-logo pointer" id="logo" src="./images/home-button.png" alt="Take Me Home">
         <menu class="nav-links">
-            <input type="text" placeholder="Search Recipes..." class="search-bar" id="searchBar">
+            <label for="searchBar" class=".off-page">Search Recipes</label>
```
Comment (jsahim) [inline dist/index.html:14]: Good call on adding a label.

**Review by jsahim — APPROVED** [review]
This looks good to me!

File: dist/index.html (hunk 1 above)
Comment (jfogiato) [reply dist/index.html:14]: Happy to have this accessibility issue resolved! Good work!

File: src/scripts.js [hunk 2]
```diff
@@ -51,16 +51,15 @@ apiCalls().then(data => {
...
+        if (recipe.name.length > 17) titleClass = 'recipe-title long-title';
+        let heartClass = "heart-icon";
+        if(!recipe.saved) heartClass = "heart-icon hidden";
         recipeSection.innerHTML += `
         <article class="recipe-card pointer" data-parent="${recipe.id}">
-            <img class="recipe-img" src="${recipe.image}" data-parent="${recipe.id}" alt="picture of ${recipe.name}">
-            <img class="${hidden}" data-parent="${recipe.id}" src="./images/heart.png" alt="This recipe is in my recipes!">
-            <h3 class="recipe-title" style="font-size: ${size}rem" data-parent="${recipe.id}">${recipe.name}</h3>
+            <img class="recipe-img" src="${recipe.image}" data-parent="${recipe.id}" alt="Picture of ${recipe.name}">
+            <img class="${heartClass}" data-parent="${recipe.id}" src="./images/heart.png" alt="This recipe is in my recipes!">
```
Comment (jfogiato) [inline src/scripts.js:61]: Good call renaming these variables. Much easier to read!

File: src/styles.css [hunk 3]
```diff
@@ -117,7 +117,7 @@ body {
 }
 
 .recipe-card {
-  height: 50vmin;
+  height: 46vmin;
```
Comment (jfogiato) [inline src/styles.css:120]: smoll

File: src/styles.css [hunk 4]
```diff
@@ -152,8 +152,24 @@ body {
...
+  margin: 0;
+  padding: 3vmin;
+  font-size: 1.6rem;
+  width: 40vmin;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+}
+
+.long-title:hover {
```
Comment (jfogiato) [inline src/styles.css:165]: HUGE work on this fix. Great work

**Review by jfogiato — APPROVED** [review]

**Review by LordSchwifty — APPROVED** [review]
nice change on the input label

**Pull request merged**